staging/src/k8s.io/kubectl/pkg/explain/explain.go

```diff
@@ -45,8 +46,20 @@ func splitDotNotation(model string) (string, []string) {
 // SplitAndParseResourceRequest separates the users input into a model and fields
 func SplitAndParseResourceRequest(inResource string, mapper meta.RESTMapper) (string, []string, error) {
 	inResource, fieldsPath := splitDotNotation(inResource)
-	inResource, _ = mapper.ResourceSingularizer(inResource)
-	return inResource, fieldsPath, nil
+	gvr, err := mapper.ResourceFor(schema.GroupVersionResource{Resource: inResource})
+	if err != nil {
+		return "", nil, err
+	}
+	singular, err := mapper.ResourceSingularizer(gvr.Resource)
+	if err != nil {
+		return "", nil, err
+	}
+	if len(gvr.Group) == 0 {
+		return singular, fieldsPath, nil
+	}
+	// Only returning the resource name is not enough. CRDs may have the same resource name.
+	// TODO(knight42): not sure whether we need to speicfy the version here.
+	return fmt.Sprintf("%s.%s", singular, gvr.Group), fieldsPath, nil
```

Comment:
why not changing this and return GroupVersionResource, fieldsPath and error here?

Reply:
How this will work in the situation where I specify `--api-version` that is different than what you discovered here? Should there be an error in explain flow? I'd expect one.

Reply:
> How this will work in the situation where I specify --api-version that is different than what you discovered here

I would like to respect the given apiVersion, since users are likely to know what they are doing if they specify `--api-version` explicitly.

Reply:
> why not changing this and return GroupVersionResource, fieldsPath and error here?

fixed